Content.Server/Database/ServerDbManager.cs

```diff
@@ -86,131 +48,360 @@ public void Init()
                 builder.AddProvider(_msLogProvider);
             });
 
+            ServerDbContext = CreateDbContext();
```

Comment:
You should absolutely not keep the DB context around for postgres. EFCore does a lot of caching internally which *will* be a problem with concurrent DB modifications from the other game servers. This isn't a problem with SQLite because there's exactly one user of the database, but absolutely not for postgres.

Reply:
Your changes also make this not thread safe. For postgres we just created separate DB contexts which are automatically thread safe. For SQLite we had a semaphore to keep a single thread accessing the DB at once. You got rid of both which is bad.